mrp_production_hierarchy/models/mrp_production.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ABF OSIELL <http://osiell.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class MrpProduction(models.Model):
+    _inherit = 'mrp.production'
+    _parent_name = "parent_id"
+    _parent_store = True
+    _parent_order = 'name'
+
+    root_id = fields.Many2one(
+        'mrp.production', u"Root order",
+        index=True, ondelete='restrict', readonly=True)
+    parent_id = fields.Many2one(
+        'mrp.production', u"Parent order",
+        index=True, ondelete='restrict', readonly=True)
+    child_ids = fields.One2many(
+        'mrp.production', 'parent_id', u"Child orders")
+    parent_left = fields.Integer('Left Parent', index=True)
+    parent_right = fields.Integer('Right Parent', index=True)
+
+    @api.multi
+    def _generate_moves(self):
```

Comment:
This is an api.multi. So self is likely to be a recordset, and in the  original method it iterates on that recordset.

IMHO you should iterate through the recordset,like:

for mo in self:
   if not mo.env.context.get('root_mrp_production_id'):
            mo = mo.with_context(root_mrp_production_id=mo.id)
   mo = mo.with_context(parent_mrp_production_id=mo.id)
   return super(MrpProduction, mo)._generate_moves()

Reply:
Thanks, fixed.